src/App.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { CheckBox } from 'react-native-elements';
+import {
+  StyleSheet,
+  View,
+  TextInput,
+  Keyboard,
+  Platform,
+  AsyncStorage,
+  FlatList,
+  Button
+} from "react-native";
+
+const isAndroid = Platform.OS == "android";
+const viewPadding = 10;
+
+type ITaskProps = Array<{ key: string, text: string, isCompleted?: boolean }>;
+
+export interface ITodoListState {
+  tasks: ITaskProps;
+  text: string;
+  viewPadding?: string | number;
+}
+
+export default class App extends React.Component<{}, ITodoListState> {
+
+  state = {
+    tasks: [],
+    text: "",
+    viewPadding: viewPadding
+  }
+
+  async componentDidMount() {
+    Keyboard.addListener(
+      isAndroid ? "keyboardDidShow" : "keyboardWillShow",
+      e => this.setState({ viewPadding: e.endCoordinates.height + viewPadding })
+    );
+
+    Keyboard.addListener(
+      isAndroid ? "keyboardDidHide" : "keyboardWillHide",
+      () => this.setState({ viewPadding: viewPadding })
+    );
+
+    const tasks = await this.getTasksFromStorage()
+    this.setState({tasks});
```

Comment:
setState only once, line 36,41 and 45 can be combined as
`this.setState({ viewPadding: someVarHoldingTheValue, tasks })`

Reply:
Setting different padding when the keyboard is visible and when it's hidden. So, this can not be combined.